sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/serializer/JsonSerializer.java

```diff
@@ -12,6 +12,17 @@
  * Generic interface covering basic JSON serialization and deserialization methods.
  */
 public interface JsonSerializer extends ObjectSerializer {
+    /**
+     * Reads a JSON stream into its object representation.
+     *
+     * @param stream JSON stream.
+     * @param typeReference {@link TypeReference} representing the object.
+     * @param <T> Type of the object.
+     * @return The object represented by the deserialized JSON stream.
+     */
+    @Override
+    <T> T deserializeSync(InputStream stream, TypeReference<T> typeReference);
```

Comment:
Probably should inverse the naming for sync and async the way we do for clients. `deserialize` and `deserializerAsync`.

Reply:
My thought process here was we want the asynchronous APIs to be the main scenario, so I decided to give the synchronous APIs the worse name.

@JonathanGiles, thoughts as well?

Reply:
Yeah, I would make it Async - it looks a little odd with the Sync suffix.

Reply:
Updated naming to use `Async` suffix for asynchronous methods, synchronous methods have no suffix.